### ISSUE TITLE
false warning: comparison is useless due to type limits

### ISSUE THREAD

**Issue by jxs** [rust-lang/rust#98808]
hey all,

I tried this code:

```rust
    let mut p: usize = 0;
    while p >= 0 && p < 10 {
        dbg!(p);
        p +=1;
    }
```
available [here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=60db89b19176bad3ba415e3b70258997)
I expected to not get the warning: `comparison is useless due to type limits`. 
Replacing `>=` with `>` or `0` with `1` solves the warning, but the behavior is different as the loop isn't entered. 

### Meta

`rustc --version --verbose`:
```
rustc 1.62.0 (a8314ef7d 2022-06-27)
binary: rustc
commit-hash: a8314ef7d0ec7b75c336af2c9857bfaf43002bfc
commit-date: 2022-06-27
host: x86_64-unknown-linux-gnu
release: 1.62.0
LLVM version: 14.0.5
```

**Comment by ChrisDenton** [comment]
It's trying to say that `p < 10` must always be `>= 0` because a `usize` is never less than zero.

So the intended suggestion is:

```rust
let mut p: usize = 0;
while p < 10 {
    dbg!(p); p +=1;
}
```

**Comment by jxs** [comment]
ah right! now that you have pointed out, it's painfully clear.  
thanks Chris!
